### PR TITLE
ion_phys/common.py: Fix calc_M1 function

### PR DESCRIPTION
**Pull request by RHanley1**
I have tried to extract the M1 matrix elements using the ```calc_M1() ``` function using the following MWE:
```python 
ion = Ca43()
ion.setB(11e-4)
ion.calc_M1()
```
However the function crashes (see below).

```python 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-0703e4df7be4> in <module>
      2 ion = Ca43()
      3 ion.setB(11e-4)
----> 4 ion.calc_M1()

c:\ion_phys\ion_phys\common.py in calc_M1(self)
    452 
    453         for level, data in self.levels.items():
--> 454             lev = level.slice()
    455             J_dim = np.rint(2.0*level.J+1).astype(int)
    456             dim = J_dim*I_dim

AttributeError: 'Level' object has no attribute 'slice'
```
These bugs have been fixed. I have also added an example script to calculate the M1 matrix elements for all transitions which was used to check against TPH thesis Table E.4.

### CODE REVIEW

File: ion_phys/common.py [hunk 1]
```diff
@@ -451,7 +451,7 @@ def calc_M1(self):
         eyeI = np.identity(I_dim)
 
         for level, data in self.levels.items():
-            lev = level.slice()
+            lev = data.slice()
```
Comment (hartytp) [inline ion_phys/common.py:454]: Thanks for fixing. The variable names are all over the place here (`lev`, `level`, `data`) and make the code somewhat hard to follow (my fault IIRC). I'd happily accept a PR at some point that made this a bit more civilized if you can think of better terminology...

File: ion_phys/examples/M1_transitions.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from ion_phys.ions.ca43 import Ca43, ground_level
+import scipy.constants as sc
+
+uB = sc.physical_constants['Bohr magneton'][0]
+
+
+if __name__ == '__main__':
+    # all seems correct (e.g. agrees with TPH thesis table E.4)
+    ion = Ca43(level_filter=[ground_level])
+    ion.setB(146.0942e-4)
```
Comment (hartytp) [inline ion_phys/examples/M1_transitions.py:10]: I wonder if we should consider making the clock qubit fields constants that can be imported from Ca43?

File: ion_phys/examples/M1_transitions.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    R = ion.M1/uB
+
+    for M3 in range(-3, +3 + 1):
+        for q in [-1, 0, 1]:
+            F4 = ion.index(ground_level, M3-q, F=4)
+            F3 = ion.index(ground_level, M3, F=3)
+            Rnm = R[ion.index(ground_level, M=M3, F=3),
+                    ion.index(ground_level, M=M3-q, F=4)]
+            print('Rnm for F=3, M={} -> F=4,' \
```
Comment (hartytp) [inline ion_phys/examples/M1_transitions.py:21]: Slash is not needed here.

File: ion_phys/common.py (hunk 1 above)
Comment (RHanley1) [reply ion_phys/common.py:454]: Yes I agree, it was a bit hard to try and work out what was going on to start with. Another slightly confusing thing is the multiple definitions of the slice() function, of which one is native to python.

Comment (hartytp) [reply ion_phys/common.py:454]: Yes, I wasn't totally happy with `slice` function name among a few others (e.g. `delta`) and would be very open to suggestions for better replacements.
